python/common/org/python/types/Str.java

```diff
@@ -282,24 +282,35 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
             if (index instanceof org.python.types.Slice) {
                 org.python.types.Slice.ValidatedValue slice = ((org.python.types.Slice) index).validateValueTypes();
                 java.lang.String sliced;
-
                 if (slice.start == null && slice.stop == null && slice.step == null) {
                     sliced = this.value;
                 } else {
                     long start;
                     if (slice.start != null) {
-                        start = toPositiveIndex(slice.start.value);
+                        if (slice.start.value < 0) {
```

Comment:
I know the original logic flow was like this, and we might not really want to touch 'working' code, but I think we can take this opportunity to make this section of code a bit cleaner since it's looking a bit more involved... Declare and instantiate `start` to 0 immediately above, since that's the 'default' value. Then that removes the need for the `else` block below.

Similarly in the rest of this function.